src/core/repr.ml

```diff
@@ -158,19 +158,24 @@ let mk_block addr tag data =
   { addr; tag; data; }
 
 
+(* When running in js_of_ocaml the representation of values differs slightly *)
+let in_jsoo = Sys.backend_type = Sys.Other "js_of_ocaml"
```

Comment:
Rather than just treating `jsoo` differently, i think it would be better to only inspect closures when we know it is safe to do so, which means only inspecting closures when the backend is bytecode or native (and/or later with specific code for exotic backends if we want to). That will probably avoid having to do the same when compilation to wasm becomes more common.

Reply:
That's a good point. I've pushed a new version that should be closer to what you suggest